### PR TITLE
continue #4453

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: src/nvim/ex_cmds2.c [hunk 1]
```diff
@@ -50,13 +50,6 @@
 #include "nvim/api/private/defs.h"
 
 
-#ifdef WIN32
-#define os_strtok_r strtok_s
-#else
-#define os_strtok_r strtok_r
-#endif
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: Moved to os_defs.h

File: src/nvim/ex_cmds2.c [hunk 2]
```diff
@@ -452,8 +441,6 @@ static char_u   *debug_skipped_name;
 /// executed.  Return true and set breakpoint_name for skipped commands that
 /// decide to execute something themselves.
 /// Called from do_one_cmd() before executing a command.
-///
-/// @param eap
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: Avoid empty param docstrings

File: src/nvim/ex_cmds2.c [hunk 3]
```diff
@@ -493,12 +480,10 @@ void dbg_check_breakpoint(exarg_T *eap)
 /// Go to debug mode if skipped by dbg_check_breakpoint() because eap->skip was
 /// set.
 ///
-/// @param eap
-///
 /// @return true when the debug mode is entered this time.
 bool dbg_check_skipped(exarg_T *eap)
 {
-  bool prev_got_int;
+  int prev_got_int;
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: Reverted this because it would result in assigning int to bool below.

File: src/nvim/ex_cmds2.c [hunk 4]
```diff
@@ -771,17 +754,17 @@ void ex_breaklist(exarg_T *eap)
 /// Returns line number at which to break; zero when no matching breakpoint.
 linenr_T
 dbg_find_breakpoint(
-    bool file,                   // true for a file, false for a function
+    bool file,             // true for a file, false for a function
     char_u *fname,         // file or function name
-    linenr_T after             // after this line number
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: Since all these lines were changed, they should be aligned.

File: src/nvim/ex_cmds2.c [hunk 5]
```diff
@@ -771,17 +754,17 @@ void ex_breaklist(exarg_T *eap)
...
+    linenr_T after         // after this line number
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: Since all these lines were changed, they should be aligned.

File: src/nvim/ex_cmds2.c [hunk 6]
```diff
@@ -2225,14 +2210,13 @@ void ex_compiler(exarg_T *eap)
   char_u      *buf;
   char_u      *old_cur_comp = NULL;
   char_u      *p;
-  size_t bufsize;
 
   if (*eap->arg == NUL) {
     // List all compiler scripts.
-    do_cmdline_cmd("echo globpath(&rtp, 'compiler//.vim')");
+    do_cmdline_cmd("echo globpath(&rtp, 'compiler/*.vim')");
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: This would have caused a bug. We must be careful about search-replace edits.

File: src/nvim/ex_cmds2.c [hunk 7]
```diff
@@ -3388,12 +3372,12 @@ static char_u **find_locales(void)
...
   // Transform locale_a string where each locale is separated by "\n"
   // into an array of locale strings.
-  loc = (char_u *)os_strtok_r((char *)locale_a, "\n", &saveptr);
+  loc = (char_u *)os_strtok((char *)locale_a, "\n", &saveptr);
 
   while (loc != NULL) {
     loc = vim_strsave(loc);
     GA_APPEND(char_u *, &locales_ga, loc);
-    loc = (char_u *)os_strtok_r(NULL, "\n", &saveptr);
+    loc = (char_u *)os_strtok(NULL, "\n", &saveptr);
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: The `os_*` wrapper functions themselves do not (should not) have a `_r` suffix.

**Comment by justinmk** [comment]
The hidden comments are all still valid. I think GitHub made inline comments even more broken than they used to be.

File: src/nvim/ex_cmds2.c [hunk 8]
```diff
@@ -2225,14 +2210,12 @@ void ex_compiler(exarg_T *eap)
   char_u      *buf;
   char_u      *old_cur_comp = NULL;
   char_u      *p;
-  size_t bufsize;
 
   if (*eap->arg == NUL) {
     // List all compiler scripts.
-    do_cmdline_cmd("echo globpath(&rtp, 'compiler//.vim')");
-    // ) keep the indenter happy...
+    do_cmdline_cmd("echo globpath(&rtp, 'compiler/*.vim')");  // NOLINT
```
Comment (justinmk) [inline src/nvim/ex_cmds2.c]: `'compiler//.vim'` would have caused a bug. We must be careful about search-replace edits.

Comment (phanimahesh) [inline src/nvim/ex_cmds2.c]: Oops, sorry, I ran the tests and had a quick look at the replacements but missed this. :/

Comment (ZyX-I) [inline src/nvim/ex_cmds2.c]: No need to NOLINT here as well, just writing `… compiler/""*.vim` needs much less characters and no linter silencing.

**Pull request merged**